docs/content/querying/dimensionspecs.md

```diff
@@ -248,6 +248,57 @@ It is illegal to set `retainMissingValue = true` and also specify a `replaceMiss
 
 A property of `injective` specifies if optimizations can be used which assume there is no combining of multiple names into one. For example: If ABC123 is the only key that maps to SomeCompany, that can be optimized since it is a unique lookup. But if both ABC123 and DEF456 BOTH map to SomeCompany, then that is NOT a unique lookup. Setting this value to true and setting `retainMissingValue` to FALSE (the default) may cause undesired behavior.
 
+A property `optimize` can be supplied to allow optimization of lookup based extraction filter (by default `optimize = false`). 
+The optimization layer will run on the broker and it will rewrite the extraction filter as clause of selector filters.
+For instance the following filter 
```

Comment:
@xvrl, as i said i don't have a strong opinion about it, my thought was, let's start with a conservative approach (off by default) and turn it on once we use it in production etc ... so if you want this to be on by default i can do it.